go/private/actions/compile.bzl

```diff
@@ -67,6 +70,8 @@ def emit_compile(go,
   args.add(go.toolchain.flags.compile)
   if go.mode.debug:
     args.add(["-N", "-l"])
+  if go.mode.link in [LINKMODE_PLUGIN]:
```

Comment:
Maybe just `go.mode.link == LINKMODE_PLUGIN` until there are other modes that require this flag. I'm guessing other modes will need additional flags, so they'll probably have their own cases.

Reply:
The intent was to have one if per flag rather than one if per mode, hence the list will include the other modes (C_SHARED and C_ARCHIVE will both do this flag).

Reply:
i'll open a PR on top of this one for C_ARCHIVE and C_SHARED if you want (they also expose headers)

Reply:
I have been working mostly on plugin mode, but hit a wall with not being able to build the standard library in a way that is ready for it.
You may hit the same problem... but feel free to try!